azurerm/internal/services/appplatform/spring_cloud_service_resource.go

```diff
@@ -618,3 +687,25 @@ func flattenArmSpringCloudGitPatternRepository(input *[]appplatform.GitPatternRe
 
 	return results
 }
+
+func flattenArmSpringCloudTrace(input *appplatform.TraceProperties) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	enabled := false
+	appInsightInstrumentationKey := ""
+	if input.Enabled != nil {
+		enabled = *input.Enabled
+	}
+	if input.AppInsightInstrumentationKey != nil {
+		appInsightInstrumentationKey = *input.AppInsightInstrumentationKey
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"enabled":                         enabled,
+			"app_insight_instrumentation_key": appInsightInstrumentationKey,
```

Comment:
(as above) we can simplify this to:

```suggestion
			"instrumentation_key": appInsightInstrumentationKey,
```